### PR TITLE
Use NSNumber for item IDs in TCRecommender

### LINKED ISSUE #832: TCRecommender uses string keys to refer to int item IDs
The TCRecommender implementation always returns a dictionary with string keys, even if the model used int item IDs. So for models trained with int item IDs, the resulting Core ML model will throw (unwrapping optional nil) when accessed from Swift, since the cast to `[Int64: Double]` will fail.

### PR DESCRIPTION
**Pull request by znation**
When the model uses int64 item IDs, we should use NSNumber as the key
type in the output dictionaries.

Fixes #832

### CODE REVIEW

File: src/objc/TuriCreateObjC.m [hunk 1]
```diff
@@ -25,6 +25,7 @@ static bool handleError(tc_error* error_ptr, NSError** error) {
                                          code:MLModelErrorGeneric
                                      userInfo:userInfo];
         }
+        assert(false);
```
Comment (shantanuchhabra) [inline src/objc/TuriCreateObjC.m]: Do we need the `assert(false)`?

Comment (TobyRoseman) [inline src/objc/TuriCreateObjC.m]: Do you want to keep this line? Is the return statement on the line below just to make the compiler happy?

Comment (znation) [reply src/objc/TuriCreateObjC.m]: I added it to debug; the return statement is probably the correct behavior here. I'll remove this line.

Comment (znation) [reply src/objc/TuriCreateObjC.m]: Probably not (see also Toby's comment). I'll remove it.

**Pull request merged**